src/main/java/blusunrize/immersiveengineering/common/util/compat/OneProbeCompatModule.java

```diff
@@ -0,0 +1,356 @@
+				max = ((IFluxReceiver) tileEntity).getMaxEnergyStored(null);
+			}
+			else if (tileEntity instanceof IFluxProvider)
+			{
+				cur = ((IFluxProvider) tileEntity).getEnergyStored(null);
+				max = ((IFluxProvider) tileEntity).getMaxEnergyStored(null);
+			}
+			if (max > 0)
+			{
+				probeInfo.progress(cur, max,
+						probeInfo.defaultProgressStyle()
+								.suffix("IF")
+								.filledColor(Lib.COLOUR_I_ImmersiveOrange)
+								.alternateFilledColor(0xff994f20)
+								.borderColor(Lib.COLOUR_I_ImmersiveOrangeShadow)
+								.numberFormat(NumberFormat.COMPACT));
+			}
+		}
+
+		@Override
+		public void getProbeConfig(
+			IProbeConfig config,
+			PlayerEntity player,
+			World world,
+			Entity entity,
+			IProbeHitEntityData data
+		)
+		{
+		}
+
+		@Override
+		public void getProbeConfig(
+			IProbeConfig config,
+			PlayerEntity player,
+			World world,
+			BlockState blockState,
+			IProbeHitData data
+		)
+		{
+			TileEntity tileEntity = world.getTileEntity(data.getPos());
+			if(tileEntity instanceof IFluxReceiver||tileEntity instanceof IFluxProvider)
+			{
+				config.setRFMode(0);				
+			}
+		}
+	}
+
+	public static class ProcessProvider implements IProbeInfoProvider
+	{
+
+		@Override
+		public String getID()
+		{
+			return ImmersiveEngineering.MODID+":"+"ProcessInfo";
+		}
+
+		@Override
+		public void addProbeInfo(
+			ProbeMode mode,
+			IProbeInfo probeInfo,
+			PlayerEntity player,
+			World world,
+			BlockState blockState,
+			IProbeHitData data
+		)
+		{
+			TileEntity tileEntity = world.getTileEntity(data.getPos());
+			if(tileEntity instanceof IEBlockInterfaces.IProcessTile)
+			{
+				int[] curTicks = ((IEBlockInterfaces.IProcessTile) tileEntity).getCurrentProcessesStep();
+				int[] maxTicks = ((IEBlockInterfaces.IProcessTile) tileEntity).getCurrentProcessesMax();
+				int h = Math.max(4, (int)Math.ceil(12/(float)curTicks.length));
+	
+				for(int i = 0; i < curTicks.length; i++)
+				{
+					if(maxTicks[i] > 0)
+					{
+						float current = curTicks[i]/(float)maxTicks[i]*100;
+						probeInfo.progress(
+							(int)current,
+							100,
+							probeInfo.defaultProgressStyle().showText(h >= 10).suffix("%").height(h)
+						);
+					}
+				}
+			}
+		}
+	}
+
+	public static class TeslaCoilProvider implements IProbeInfoProvider
+	{
+
+		@Override
+		public String getID()
+		{
+			return ImmersiveEngineering.MODID+":"+"TeslaCoilInfo";
+		}
+
+		@Override
+		public void addProbeInfo(
+			ProbeMode mode,
+			IProbeInfo probeInfo,
+			PlayerEntity player,
+			World world,
+			BlockState blockState,
+			IProbeHitData data
+		)
+		{
+			TileEntity tileEntity = world.getTileEntity(data.getPos());
+			
+			if(tileEntity instanceof TeslaCoilTileEntity)
+			{
+				TeslaCoilTileEntity teslaCoil = (TeslaCoilTileEntity) tileEntity;
+				if(teslaCoil.isDummy())
+				{
+					tileEntity = world.getTileEntity(
+							data.getPos().offset(teslaCoil.getFacing(), -1));
```

Comment:
This does not need to be split. We do not enforce line lengths (unless they are ridiculously long).